Create an interface for base SNS velocity IK solver

robodreamer opened the pull request:
This commit adds an interface that allows SNS Base IK
solver to be used as one of SNS IK solver types.
SNS_Base type has been defined and position and velocity
IK solver tests now include tests for this new type.

Commit:
Create an interface for base SNS veloicty IK solver

This commit adds an interface that allows SNS Base IK
solver to be used as one of SNS IK solver types.
SNS_Base type has been defined and position and velocity
IK solver tests now include tests for this new type.

Commit:
Fix a bug in acc_ik_base_test code

This commit fixes a bug that let unit test fail.

chris-smith commented on sns_ik_lib/src/sns_vel_ik_base.cpp:
```diff
@@ -294,7 +294,7 @@ SnsIkBase::ExitCode SnsVelIkBase::solve(const Eigen::MatrixXd& J, const Eigen::V
   }
   else if (*taskScaleCS < MINIMUM_FINITE_SCALE_FACTOR) {
-    ROS_WARN("Secondary goal is infeasible! scaling --> zero");
+    ROS_DEBUG_THROTTLE(0.5, "Secondary goal is infeasible! scaling --> zero");
```
If you're already making this debug, do you still need to throttle it? same question for other similar change

robodreamer replied:
Right. I agree. Debug will be fine.

robodreamer replied:
Thanks for your comments! :)

chris-smith commented on sns_ik_lib/src/sns_vel_ik_base_interface.cpp:
```diff
@@ -0,0 +1,98 @@
+  baseIkSolver->setBounds(dqLow, dqUpp);
+
+  // initialize the base IK solver
+  // ROS_INFO_STREAM("dqLow: " << dqLow.transpose().format(EigArrFmt));
```
remove logging?

chris-smith commented on sns_ik_lib/src/sns_vel_ik_base_interface.cpp:
```diff
@@ -0,0 +1,98 @@
+  }
+
+
+  // ROS_INFO_STREAM("taskScale: " << taskScale);
```
and here

robodreamer replied:
I forgot to remove them..

chris-smith commented:
Mostly wondering about renaming the new class? `SNSVelocityBaseIK` which inherits from `SNSVelocityIK` and has a member `SnsVelIkBase`... I have a hard time figuring out what each does from the names and not getting them mixed up.

Commit:
Adressed reviewer's comments

- renamed interface class to avoid confusion
- removed some logging for debugging

chris-smith reviewed (approved):
lgtm
